src/Init/SimpLemmas.lean

```diff
@@ -23,10 +23,10 @@ theorem eq_false (h : ¬ p) : p = False :=
 theorem eq_false' (h : p → False) : p = False :=
   propext <| Iff.intro (fun h' => absurd h' h) (fun h' => False.elim h')
 
-theorem eq_true_of_decide {p : Prop} {s : Decidable p} (h : decide p = true) : p = True :=
-  propext <| Iff.intro (fun h => trivial) (fun _ => of_decide_eq_true h)
+theorem eq_true_of_decide {p : Prop} {_ : Decidable p} (h : decide p = true) : p = True :=
+  propext <| Iff.intro (fun _ => trivial) (fun _ => of_decide_eq_true h)
 
-theorem eq_false_of_decide {p : Prop} {s : Decidable p} (h : decide p = false) : p = False :=
+theorem eq_false_of_decide {p : Prop} {_ : Decidable p} (h : decide p = false) : p = False :=
```

Comment:
I really dislike that this disables the `(s := ...)` syntax to specify implicit arguments.  (Not that `s` is a great name, `inst` or something would certainly be better.)

Reply:
> implicit-as-inst-implicit pattern

Do you mean `{_ : Decidable p}` instead of `[Decidable p]`?  (Not sure I understood you correctly.  I would have named it the way around: inst-implicit-as-implicit pattern.)

Reply:
Yes, I was thinking of it the other way around - an implicit parameter that becomes an inst-implicit argument in an application

Reply:
I just noticed that the `{_ :` style is prevalent in e.g. `Std.Data.PersistentHashMap`, so this really is a pre-existing style issue that should be discussed at a larger scope.

Reply:
Just a quick idea, but a "nicer" way could be to allow selecting parameters by type as in `eq_false_of_decide (‹Decidable p› := ...)`. One slight downside would be a larger parser overlap.